application/src/main/java/org/thingsboard/server/actors/ruleChain/DefaultTbContext.java

```diff
@@ -1010,4 +998,21 @@ private static String getFailureMessage(Throwable th) {
         return failureMessage;
     }
 
+    private void persistDebugOutput(TbMsg msg, String relationType) {
+        persistDebugOutput(msg, Set.of(relationType));
+    }
+
+    private void persistDebugOutput(TbMsg msg, Set<String> relationTypes) {
+        persistDebugOutput(msg, relationTypes, null, null);
+    }
+
+    private void persistDebugOutput(TbMsg msg, Set<String> relationTypes, Throwable error, String failureMessage) {
+        RuleNode ruleNode = nodeCtx.getSelf();
+        if (DebugModeUtil.isDebugAllAvailable(ruleNode)) {
+            relationTypes.forEach(relationType -> mainCtx.persistDebugOutput(getTenantId(), ruleNode.getId(), msg, relationType, error, failureMessage));
+        } else if (DebugModeUtil.isDebugFailuresAvailable(ruleNode, relationTypes)) {
```

Comment:
Why not `ruleNode.isDebugEnabled(relationTypes)`?

Reply:
I think this better for understanding where we should use it, 
isDebugAllAvailable - for all debug events
isDebugFailuresAvailable - only for failures
If we will rename it to isDebugEnabled, everyone will use it without relation types.